src/measure_theory/lp_space.lean

```diff
@@ -1540,19 +1540,23 @@ lemma continuous_comp_Lp [fact (1 ≤ p)] (hg : lipschitz_with c g) (g0 : g 0 =
 end lipschitz_with
 
 namespace continuous_linear_map
-variables [normed_space ℝ E] [normed_space ℝ F]
+variables {𝕜 : Type*} [is_R_or_C 𝕜] [normed_space 𝕜 E] [normed_space 𝕜 F]
 
-/-- Composing `f : Lp ` with `L : E →L[ℝ] F`. -/
-def comp_Lp (L : E →L[ℝ] F) (f : Lp E p μ) : Lp F p μ :=
+/-- Composing `f : Lp ` with `L : E →L[𝕜] F`. -/
+def comp_Lp (L : E →L[𝕜] F) (f : Lp E p μ) : Lp F p μ :=
 L.lipschitz.comp_Lp (map_zero L) f
```

Comment:
What actually needs `is_R_or_C` here? Is `[nondiscrete_normed_field 𝕜]` enough?

Reply:
This def and the two lemmas below it are true for only
```lean
variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜] [semi_normed_space 𝕜 E] [semi_normed_space 𝕜 F]
```

Reply:
Indeed all those hypotheses can be relaxed. This is code I wrote before all the `normed_space` in mathlib became `semi_normed_space`.
I'll add `semi_` to all hypotheses in the file (if that works).

Reply:
It worked without any issue.

Reply:
It's more complicated in the set_integral file. Either some `normed_space` in other files should be replaced by `semi_normed_space`, or most `normed_space` arguments are actually needed.

Reply:
Perhaps the `normed_group` arguments need to be replaced by `semi_normed_group`?  Just a thought, I haven't tested it.

Reply:
Right, I replaced some `normed_space` by `semi_normed_space`, but those are all still `normed_group` (and changing to `semi_normed_group` looks like it causes errors). So I effectively still have `normed_space` and it is as if I did not do anything. I suppose the two main possible courses of action are to keep things as they are here, or to work towards using `semi_normed_group` as much as possible.

I confess that investigating that normed vs seminormed issue in all Lp/integral files does not interest me very much right now, and I think that it is beyond the goal of this PR.

Reply:
> I confess that investigating that normed vs seminormed issue in all Lp/integral files does not interest me very much right now, and I think that it is beyond the goal of this PR.

I agree!

Reply:
Sorry for sending you into that rabbit hole - I only meant it as a throwaway observation, the change to `nondiscrete_normed_field` was much more relevant.